datafusion/substrait/src/lib.rs

```diff
@@ -15,6 +15,61 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Serialize / Deserialize DataFusion Plans to [Substrait.io]
+//!
+//! This crate provides support for serializing and deserializing DataFusion plans
+//! to and from the generated types in [substrait::proto] from the [substrait] crate.
+//!
+//! [Substrait.io] provides a cross-language serialization format for relational
+//! algebra (e.g. query plans and expressions), based on protocol buffers.
+//!
+//! [Substrait.io]: https://substrait.io/
+//!
+//! [`LogicalPlan`]: datafusion::logical_expr::LogicalPlan
+//! [`ExecutionPlan`]: datafusion::physical_plan::ExecutionPlan
+//!
+//! Potential uses of this crate:
+//! * Use DataFusion run Substrait plans created by other systems (e.g. Apache Calcite)
+//! * Use DataFusion to create plans to run on other systems
+//! * Pass query plans over FFI boundaries, such as from Python to Rust
```

Comment:
I'd like to provide another use case from GreptimeDB, where substrait plan is used to pass query plans across compute nodes.

Reply:
I added a note in  708c7cd11

I am curious why you chose to substrait and not datafusion-proto  *within* the same system (we use datafusion-proto for this case, for its wider compatibility)

Reply:
I'm considering combining other consumer/producer together so the "node" need not be the same at that time. It turns out the work is tremendous and the related ecosystem is not ready yet 🤣

Reply:
> I'm considering combining other consumer/producer together so the "node" need not be the same at that time.

That makes a lot of sense

>  It turns out the work is tremendous and the related ecosystem is not ready yet 🤣

Yeah, that is my feeling too. One reason I felt this PR would be good is that it might help find / attract people who want to play with the bleeding edge and make DataFusion's substrait support better. We'll see 🎣